source/tools/GenContainer.rst

```diff
@@ -16,39 +16,95 @@ A container is an encapsulation of multiple components as depicted in the follow
+This type of container contains a single component in it. Component format can be ELF, UEFI-PI FV, or PE32.
 
-Classic
-^^^^^^^
+ - example command::
 
-This image type is used to boot a typical Linux image. This image format combines a kernel command line and the kernel image.
-Optionally more files like an initrd, ACPI tables, firmware files could be added. 
+    python GenContainer.py create -t NORMAL -cl UEFI:UefiPld.fd
+
+2. Classic Linux
+^^^^^^^^^^^^^^^^
+
+This container type is used to boot a typical Linux operating system. This container is made up of atleast two
+components - the kernel command line, and kernel image.
+
+**SBL expects the components inside the container in the following order: kernel command line, kernel image,
+and the optional initrd.**
 
-The kernel command line, kernel image, and the optional initrd are in a **fixed order**.
   * Dummy files need to be provided if one of the component is not in use. 
   * For example, if initrd is not used, a dummy file needs to be provided in place of the initrd.
 
+Optionally, more components like ACPI tables, firmware blobs, etc. can be added to the container. These blobs
+will be loaded by SBL into memory.
+
+Some device drivers may require binary blobs for initialization. These are typically loaded from the filesytem.
+In some use cases like fast boot, having access to these blobs before the filesystem is mounted may be desirable.
+Packaging blobs with the container provides these drivers a way to get the blobs without waiting for the filesystem
+to be mounted. SBL can pass the memory address of the loaded blob through the Linux command line. For this, SBL
+expects a placeholder in the command line to put the address into. For example, if the blob's component name
+when creating the container was specified as ``BLOB``, SBL will look for the string ``SBL.BLOB=0x0000000000000000``
+in the cmdline. The zeroes will be replaced by the actual memory address at which the blob will be loaded.
+
 **File Order:** cmdline.txt, bzImage (kernel), initrd, acpi, firmware1, firmware2, ...
 
-The classic container's format is laid out below.
+The Classic Linux container's format is laid out below.
 
   .. image:: /images/sbl_classic_container.png
       :width: 600px
 
-Multiboot ELF
-^^^^^^^^^^^^^
+ - example command::
+
+    python GenContainer.py create -t CLASSIC -cl CMDL:cmdline.txt KRNL:bzImage INRD:initrd SBL.BLOB=0x0000000000000000
 
-This type of container stores Multiboot compliant ELF images and their corresponding command lines in pairs. The first ELF image \
-in the Multiboot image is assumed to be the one used for booting. If an image does not use a command line, a dummy file needs to be \
+3. Multiboot ELF
+^^^^^^^^^^^^^^^^
+
+This type of container stores Multiboot / Multiboot-2 compliant ELF images and their corresponding command line files in pairs. The first ELF image \
```

Comment:
This type of container stores Multiboot / Multiboot-2 compliant ELF images -> This type of container stores Multiboot / Multiboot-2 compliant ELF image components

Reply:
Does "ELF image components" sound like the ELF itself is being split into components?

Should we change it to "ELF binaries" instead?

Reply:
Thats fine too

Reply:
Thanks